sensor/common/filesystem/pipeline/pipeline.go

```diff
@@ -0,0 +1,173 @@
+
+	switch fs.GetFile().(type) {
+	case *sensorAPI.FileActivity_Creation:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetCreation().GetActivity().GetPath(),
+			HostPath: fs.GetCreation().GetActivity().GetHostPath(),
+		}
+		access.Operation = storage.FileAccess_CREATE
+	case *sensorAPI.FileActivity_Unlink:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetUnlink().GetActivity().GetPath(),
+			HostPath: fs.GetUnlink().GetActivity().GetHostPath(),
+		}
+		access.Operation = storage.FileAccess_UNLINK
+	case *sensorAPI.FileActivity_Rename:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetRename().GetOld().GetPath(),
+			HostPath: fs.GetRename().GetOld().GetHostPath(),
+		}
+		access.Moved = &storage.FileAccess_File{
+			Path:     fs.GetRename().GetNew().GetPath(),
+			HostPath: fs.GetRename().GetNew().GetHostPath(),
+		}
+		access.Operation = storage.FileAccess_RENAME
+	case *sensorAPI.FileActivity_Permission:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetPermission().GetActivity().GetPath(),
+			HostPath: fs.GetPermission().GetActivity().GetHostPath(),
+			Meta: &storage.FileAccess_FileMetadata{
+				Mode: fs.GetPermission().GetMode(),
+			},
+		}
+		access.Operation = storage.FileAccess_PERMISSION_CHANGE
+	case *sensorAPI.FileActivity_Ownership:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetOwnership().GetActivity().GetPath(),
+			HostPath: fs.GetOwnership().GetActivity().GetHostPath(),
+			Meta: &storage.FileAccess_FileMetadata{
+				Uid:      fs.GetOwnership().GetUid(),
+				Gid:      fs.GetOwnership().GetGid(),
+				Username: fs.GetOwnership().GetUsername(),
+				Group:    fs.GetOwnership().GetGroup(),
+			},
+		}
+		access.Operation = storage.FileAccess_OWNERSHIP_CHANGE
+	case *sensorAPI.FileActivity_Write:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetWrite().GetActivity().GetPath(),
+			HostPath: fs.GetWrite().GetActivity().GetHostPath(),
+		}
+		access.Operation = storage.FileAccess_WRITE
+	case *sensorAPI.FileActivity_Open:
+		access.File = &storage.FileAccess_File{
+			Path:     fs.GetOpen().GetActivity().GetPath(),
+			HostPath: fs.GetOpen().GetActivity().GetHostPath(),
+		}
+		access.Operation = storage.FileAccess_OPEN
+	default:
+		log.Warn("Not implemented file activity type")
+		return nil
+	}
+
+	return access
+}
+
+func (p *Pipeline) getIndicator(process *sensorAPI.ProcessSignal) *storage.ProcessIndicator {
+	pi := &storage.ProcessIndicator{
+		Id: uuid.NewV4().String(),
+		Signal: &storage.ProcessSignal{
+			Id:           process.GetId(),
+			Uid:          process.GetUid(),
+			Gid:          process.GetGid(),
+			Time:         process.GetCreationTime(),
+			Name:         process.GetName(),
+			Args:         process.GetArgs(),
+			ExecFilePath: process.GetExecFilePath(),
+			Pid:          process.GetPid(),
+			Scraped:      process.GetScraped(),
+			ContainerId:  process.GetContainerId(),
+		},
+	}
+
+	if process.GetContainerId() != "" {
```

Comment:
The logic here can be simplified.

if process.GetContainerId() == "" {
// add node fields to pi and return

}

//fill in deployment fields to pi and return